packages/components/src/molecules/ValidationOverview/utils.tsx

```diff
@@ -103,29 +104,33 @@ export const filterProblems = (
   filters: ValidationFiltersValueType,
   securityFrameworkFilter: string
 ) => {
-  const suppressedFilter = filters.showSuppressed
-    ? () => true
-    : (problem: ValidationResult) => !problem.suppressions?.length;
-  const unsuppressedFilter = filters.showUnsuppressed
-    ? () => true
-    : (problem: ValidationResult) => problem.suppressions?.length;
-
   return Object.fromEntries(
     Object.entries(problems || {})
       .map(([filePath, validationResults]) => {
-        let filteredValidationResults = validationResults
-          .filter(suppressedFilter)
-          .filter(unsuppressedFilter)
-          .filter(
-            el =>
-              (filters['type'] ? el.level === filters['type'] : true) &&
-              (filters['tool-component']?.length
-                ? filters['tool-component'].includes(el.rule.toolComponent.name)
-                : true) &&
-              (securityFrameworkFilter !== 'all'
-                ? el.taxa?.find(t => t.toolComponent.name === securityFrameworkFilter)
-                : true)
-          );
+        let filteredValidationResults = validationResults.filter(
```

Comment:
If you care to do it, I think we could greatly reduce complexity here by using `.reduce()`

Reply:
Bit short on time so I'll leave this for future me 🙈